### PR TITLE
NIAD-5: enhance the documentation of monitoring and audit

### PR DESCRIPTION
**Pull request by Broom94**

### CODE REVIEW

File: mhs/operating-mhs-adaptor.md [hunk 1]
```diff
@@ -50,21 +50,24 @@ Log messages produced by the MHS Adaptor are in human readable form and importan
 and used for searching, filtering, agregation, graphing etc.
 
 ```text
-[2019-10-10T10:10:00.735207Z] Entered async express workflow to handle outbound message RequestId=459D9095-45F2-4C2D-86EA-E1E0F78BC142 CorrelationId=12345 ProcessKey=ASYNC_EXPRESS_WORKFLOW0001 pid=6 LogLevel=INFO LoggerName=ASYNC_EXPRESS_WORKFLOW
+2020-03-26T10:31:52.118165Z | AUDIT | 40186 | QUPC_IN160101UK05 | 4DE4BBB7-F1DE-48BD-8824-E8FFCE4FC703 | 1 |  | outbound.mhs_common.workflow.asynchronous_express | WorkflowName=async-express outbound workflow invoked.
```
Comment (bartek-sarul) [inline mhs/operating-mhs-adaptor.md:60]: I think it's worth adding the pattern itself:

```
"[%(asctime)sZ] | %(levelname)s | %(process)d | %(interaction_id)s | %(message_id)s | %(correlation_id)s | %(inbound_message_id)s | %(name)s | %(message)s"
```

and explain that the one missing in your example is the `inbound_message_id`

File: mhs/operating-mhs-adaptor.md [hunk 2]
```diff
@@ -50,21 +50,24 @@ Log messages produced by the MHS Adaptor are in human readable form and importan
...
+| `pid` | Process ID, identifies the running instance of a component that produced the log message. |
+| `InteractionId` | This ID is used to determine the workflow the message will take, there our four different workflows sync, sync-async, express and forward reliable. |
+| `MessageId` | A unique ID which is generated at the very start of a workflow if not already assigned. |
+| `CorrelationId` | A unique ID which is generated at the very start of a workflow and is used throughout all log messages related to that work, such that all the logs in the chain can be tied together for a single work item. CorrelationId can be passed into the MHS components from the supplier's calling client to allow for the CorrelationId to also tie the workflow together with the client system. |
+| `RefToMessageId` | A unique ID which is generated at the very start of a workflow if not already assigned. UUID assigned to incoming inbound messages that refer to original message_id of outbound message. Used to find if message id exists in state database to determine if message is unsolicited or not. |
 | `LoggerName` | Identifies the sub-component within the solution which produced the logs. |
```
Comment (bartek-sarul) [inline mhs/operating-mhs-adaptor.md]: Worth mentioning it's a dot-separated name where the first part is the component that produced the log.

File: mhs/operating-mhs-adaptor.md [hunk 3]
```diff
@@ -50,21 +50,24 @@ Log messages produced by the MHS Adaptor are in human readable form and importan
 and used for searching, filtering, agregation, graphing etc.
```
Comment (bartek-sarul) [inline mhs/operating-mhs-adaptor.md]: Might be good to mention that few first logs don't follow the pattern as they are written before logger initialization. Those are minor read-config logs

**Review by bartek-sarul — CHANGES_REQUESTED** [review]
please check my comments

**Review by grzegorz-robak — DISMISSED** [review]

File: mhs/operating-mhs-adaptor.md [hunk 4]
```diff
@@ -65,8 +73,8 @@ and used for searching, filtering, agregation, graphing etc.
...
 | `CorrelationId` | A unique ID which is generated at the very start of a workflow and is used throughout all log messages related to that work, such that all the logs in the chain can be tied together for a single work item. CorrelationId can be passed into the MHS components from the supplier's calling client to allow for the CorrelationId to also tie the workflow together with the client system. |
-| `RefToMessageId` | A unique ID which is generated at the very start of a workflow if not already assigned. UUID assigned to incoming inbound messages that refer to original message_id of outbound message. Used to find if message id exists in state database to determine if message is unsolicited or not. |
-| `LoggerName` | Identifies the sub-component within the solution which produced the logs. |
+| `Inbound Message Id` | Also known as `RefToMessageId` A unique ID which is generated at the very start of a workflow if not already assigned. UUID assigned to incoming inbound messages that refer to original message_id of outbound message. Used to find if message id exists in state database to determine if message is unsolicited or not. |
```
Comment (bartek-sarul) [inline mhs/operating-mhs-adaptor.md]: Double check that with this:
https://github.com/nhsconnect/integration-adaptors/blob/develop/mhs/inbound/inbound/request/handler.py#L55

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: mhs/operating-mhs-adaptor.md (hunk 2 above)
Comment (michael-kainos) [reply mhs/operating-mhs-adaptor.md]: Is the description for the InteractionId correct, i thought that the workflows were message patterns and interactions were relating to the spine request used?

Comment (Broom94) [reply mhs/operating-mhs-adaptor.md]: InteractionId determines the workflow used

Comment (grzegorz-robak) [reply mhs/operating-mhs-adaptor.md]: It's a little misleading. We have defined defaults for each InteractionId - it's not like InteractionId is the value that decides which workflow to use

**Review by michael-kainos — DISMISSED** [review]

**Comment by Alicja-Kainos** [comment]
> please check my comments

All comments has been checked and appropriate changes made

**Review by bartek-sarul — APPROVED** [review]

**Pull request merged**